ENIGMAsystem/SHELL/Widget_Systems/Win32/dialogs.cpp

```diff
@@ -192,12 +198,208 @@ static INT_PTR CALLBACK ShowMessageExtProc(HWND hwndDlg, UINT uMsg, WPARAM wPara
   return 0;
 }
 
-static INT CALLBACK GetDirectoryAltProc(HWND hwnd, UINT uMsg, LPARAM lp, LPARAM pData)
+WCHAR wstrPromptStr[4096];
+WCHAR wstrTextEntry[MAX_PATH];
+WCHAR wstrWindowStr[MAX_PATH];
+bool HideInput = 0;
+
+void ClientResize(HWND hWnd, int nWidth, int nHeight)
 {
-  if (uMsg == BFFM_INITIALIZED)
-    SetWindowText(hwnd, gs_cap.c_str());
+  RECT rcClient, rcWind;
+  POINT ptDiff;
+  GetClientRect(hWnd, &rcClient);
+  GetWindowRect(hWnd, &rcWind);
+  ptDiff.x = (rcWind.right - rcWind.left) - rcClient.right;
+  ptDiff.y = (rcWind.bottom - rcWind.top) - rcClient.bottom;
+  MoveWindow(hWnd, rcWind.left, rcWind.top, nWidth + ptDiff.x, nHeight + ptDiff.y, TRUE);
+}
 
-  return 0;
+LRESULT CALLBACK InputBoxHookProc(HWND hWndDlg, UINT Msg, WPARAM wParam, LPARAM lParam)
+{
+  switch (Msg)
+  {
+  case WM_INITDIALOG:
+    ClientResize(hWndDlg, 357, 128);
+    RECT rect;
+    GetWindowRect(hWndDlg, &rect);
+    MoveWindow(hWndDlg,
+      (GetSystemMetrics(SM_CXSCREEN) / 2) - ((rect.right - rect.left) / 2),
+      (GetSystemMetrics(SM_CYSCREEN) / 3) - ((rect.bottom - rect.top) / 3),
+      rect.right - rect.left, rect.bottom - rect.top, TRUE);
+    MoveWindow(GetDlgItem(hWndDlg, IDOK), 272, 10, 75, 23, TRUE);
```

Comment:
Why is this not part of the `.rc` file? This is a ton of magic numbers that people have to look at when they go to edit this file.

Reply:
Size and positioning is not measured in pixels in RC files. I need it to be in exact pixels for what I am imitating, because I do not know the measurement method RC files use or how to convert pixels to that format. Plus, there is no "MoveWindow" in an RC file. I need that to make the dialog go 1/3 of the screen down on the Y-axis.

Reply:
I suspect that the RC files use relative units, specifically to address this concern.